skimage/registration/_masked_phase_cross_correlation.py

```diff
@@ -1,303 +1,304 @@
+    .. [2] D. Padfield. "Masked FFT registration". In Proc. Computer Vision and
+           Pattern Recognition, pp. 2918-2925 (2010).
+           :DOI:`10.1109/CVPR.2010.5540032`
+
+    """
+    if moving_mask is None:
+        if reference_image.shape != moving_image.shape:
+            raise ValueError(
+                "Input images have different shapes, moving_mask must "
+                "be explicitely set.")
+        moving_mask = reference_mask.astype(bool)
+
+    # We need masks to be of the same size as their respective images
+    for (im, mask) in [(reference_image, reference_mask),
+                       (moving_image, moving_mask)]:
+        if im.shape != mask.shape:
+            raise ValueError(
+                "Image sizes must match their respective mask sizes.")
+
+    xcorr = cross_correlate_masked(moving_image, reference_image,
+                                   moving_mask, reference_mask,
+                                   axes=tuple(range(moving_image.ndim)),
+                                   mode='full',
+                                   overlap_ratio=overlap_ratio)
+
+    # Generalize to the average of multiple equal maxima
+    maxima = np.stack(np.nonzero(xcorr == xcorr.max()), axis=1)
+    center = np.mean(maxima, axis=0)
+    shifts = center - np.array(reference_image.shape) + 1
+
+    # The mismatch in size will impact the center location of the
+    # cross-correlation
+    size_mismatch = (np.array(moving_image.shape)
+                     - np.array(reference_image.shape))
+
+    return -shifts + (size_mismatch / 2)
+
+
+def cross_correlate_masked(arr1, arr2, m1, m2, mode='full', axes=(-2, -1),
+                           overlap_ratio=0.3):
+    """
+    Masked normalized cross-correlation between arrays.
+
+    Parameters
+    ----------
+    arr1 : ndarray
+        First array.
+    arr2 : ndarray
+        Seconds array. The dimensions of `arr2` along axes that are not
+        transformed should be equal to that of `arr1`.
+    m1 : ndarray
+        Mask of `arr1`. The mask should evaluate to `True`
+        (or 1) on valid pixels. `m1` should have the same shape as `arr1`.
+    m2 : ndarray
+        Mask of `arr2`. The mask should evaluate to `True`
+        (or 1) on valid pixels. `m2` should have the same shape as `arr2`.
+    mode : {'full', 'same'}, optional
+        'full':
+            This returns the convolution at each point of overlap. At
+            the end-points of the convolution, the signals do not overlap
+            completely, and boundary effects may be seen.
+        'same':
+            The output is the same size as `arr1`, centered with respect
+            to the `‘full’` output. Boundary effects are less prominent.
+    axes : tuple of ints, optional
+        Axes along which to compute the cross-correlation.
+    overlap_ratio : float, optional
+        Minimum allowed overlap ratio between images. The correlation for
```

Comment:
```suggestion
        Minimum allowed overlap ratio between the two arrays. The correlation for
```
We hadn't used the word 'image' so far.